beep/featurize.py

```diff
@@ -676,38 +849,13 @@ def features_from_processed_cycler_run(cls, processed_cycler_run, params_dict=No
         X = pd.DataFrame(np.zeros((1, 42)))
 
         # Determine beginning and end of investigated cycle type
-        index_pos_list = [
-            i
-            for i in range(len(diagnostic_interpolated.cycle_type))
-            if diagnostic_interpolated.cycle_type[i]
-            == params_dict["diagnostic_cycle_type"]
-        ]
+        index_pos_list = np.where(
+            diagnostic_interpolated.cycle_type == params_dict["diagnostic_cycle_type"])[0]
```

Comment:
Would it be better to use `np.asarray(condition).nonzero()`?

Reply:
I personally like np.where because I think it's more readable, but I see in the numpy docs that they suggest calling nonzero directly, so I changed it.